src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -184,7 +184,7 @@ protected function toNumber($value)
 
     protected function toInteger($value)
     {
-        if(ctype_digit ($value)) {
+        if(is_numeric($value) && (string)(int)$value == $value) {
```

Comment:
Not totally sure of the implications, to be honest. I suppose at some point I should cover all the edges in the tests, but for now I removed the extra cast and the existing tests still pass.